### DISCUSSION TITLE
Setting a rust version (to 1.80)

### DISCUSSION THREAD

**Discussion by rtyler** [delta-io/delta-rs#2837] [Ideas]
:warning: I propose for the deltalake 1.0 release we default our rust version to **1.80** _and_ we modify our GitHub Actions to only use the _defined_ version in `Cargo.toml`. :warning: 

I think 1.80 is a good release ("_they're all good releases_" :dog:) but would serve as a strong baseline we can reliably publish a 1.0 stable series of releases with.

---

AWS pushed new SDK crates yesterday which brought their minimum rust version up to 1.78 ([example](https://crates.io/crates/aws-sdk-ssooidc/versions))

While investigating the CI errors, I noted that our current `rust-version` is `1.75` .. which doesn't actually work, and hasn't for some time 

## :x: 1.75
```
cargo +1.75.0 build
error: package `datafusion-functions v41.0.0` cannot be built because it requires rustc 1.76 or newer, while the currently active rustc version is 1.75.0
Either upgrade to rustc 1.76 or newer, or use
cargo update datafusion-functions@41.0.0 --precise ver
where `ver` is the latest version of `datafusion-functions` supporting rustc 1.75.0
```

:face_with_head_bandage: oooooppps!

## :x: 1.76

```
cargo +1.76.0 build
error: unsupported output in build script of `delta_kernel v0.3.0`: `cargo::rustc-check-cfg=cfg(NIGHTLY_CHANNEL)`
Found a `cargo::key=value` build directive which is reserved for future use.
Either change the directive to `cargo:key=value` syntax (note the single `:`) or upgrade your version of Rust.
See https://doc.rust-lang.org/cargo/reference/build-scripts.html#outputs-of-the-build-script for more information about build script outputs.
warning: build failed, waiting for other jobs to finish...
```

:face_with_head_bandage: oooooppps!

## :x:  1.77

```
cargo +1.77.0 build
error: package `aws-sdk-dynamodb v1.43.0` cannot be built because it requires rustc 1.78.0 or newer, while the currently active rustc version is 1.77.0
Either upgrade to rustc 1.78.0 or newer, or use
cargo update aws-sdk-dynamodb@1.43.0 --precise ver
where `ver` is the latest version of `aws-sdk-dynamodb` supporting rustc 1.77.0
````

## :green_apple:  1.78

The entire system builds on 1.78 later

**Comment by ion-elgreco** [comment]
Lets go with a minimum of 1.8!

**Comment by rtyler** [comment]
Pull request in #2842 sets the `Cargo.toml` and ensures some of our GitHub Actions CI rely on that version
